File: demos/api/viewlist.cpp
#include <gmsh.h>

int main(int argc, char **argv)
{
  gmshInitialize();
  gmshOptionSetNumber("General.Terminal", 1);

  std::vector<double> tri1 = {0., 1., 1.,
                              0., 0., 1.,
                              0., 0., 0.};
  std::vector<double> tri2 = {0., 1., 0.,
                              0., 1., 1.,
                              0., 0., 0.};

  for(int step = 0; step < 10; step++){
    tri1.push_back(10.);
    tri1.push_back(10.);
    tri1.push_back(12. + step);
    tri2.push_back(10.);
    tri2.push_back(12. + step);
    tri2.push_back(13. + step);
  }

  int t = gmshViewCreate("some data");
  std::vector<double> data;
  data.insert(data.end(), tri1.begin(), tri1.end());
  data.insert(data.end(), tri2.begin(), tri2.end());

  gmshViewAddListData(t, "ST", 2, data);

  gmshViewExport(t, "data.pos");

  gmshFinalize();
  return 0;
}

File: demos/api/viewlist.py
#!/usr/bin/env python

import gmsh
import sys

gmsh.initialize(sys.argv)
gmsh.optionSetNumber("General.Terminal", 1)

tri1 = [0., 1., 1.,
        0., 0., 1.,
        0., 0., 0.];
tri2 = [0., 1., 0.,
        0., 1., 1.,
        0., 0., 0.];

for step in range(0, 10):
    tri1.append(10.); tri1.append(10.); tri1.append(12. + step)
    tri2.append(10.); tri2.append(12. + step); tri2.append(13. + step)

t = gmsh.viewAdd("some data")

gmsh.viewAddListData(t, "ST", 2, tri1 + tri2)

gmsh.viewWrite(t, "data.pos")

gmsh.finalize()
